biz.aQute.bndlib/src/aQute/bnd/service/reporter/ReportTransformerPlugin.java

```diff
@@ -0,0 +1,47 @@
+package aQute.bnd.service.reporter;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * This plugin transform an input model by applying a template on it.
+ */
+@ProviderType
+public interface ReportTransformerPlugin {
+
+	/**
+	 * Get the set of file extension names corresponding to the template format
+	 * that this plugin support.
+	 * 
+	 * @return one or multiple extensions name, never {@code null}
+	 */
+	public String[] getHandledTemplateExtensions();
+
+	/**
+	 * Get the set of file extension names corresponding to the model format
+	 * that this plugin support.
+	 * 
+	 * @return one or multiple extensions name, never {@code null}
+	 */
+	public String[] getHandledModelExtensions();
+
+	/**
+	 * Transform the model by applying the template on it and write the result
+	 * to the output stream.
+	 * 
+	 * @param model an input stream that contains the model, must not be
+	 *            {@code null}
+	 * @param template an input stream that contains the template, must not be
+	 *            {@code null}
+	 * @param output the output stream to write the transformation result, must
+	 *            not be {@code null}
+	 * @param parameters a map of parameters and their value that must be
```

Comment:
This is not a very type safe way to pass parameter. You might be better of accepting an Object argument which is a DTO type  defined by the template engines to pass parameters in a type safe way.

Reply:
The parameters are not for the template engine itself, but to be inserted as parameter in the template file for this specific transformation. So the type of the parameters are handled by the template engine that read the template file. For xslt, parameters correspond to `<xsl:param>` for the jtwig engine there is no concept of parameters, so this map is added to the data model for this specific transformation.

Parameters are defined by the user in bnd.bnd files.